packages/block-library/src/post-template/index.php

```diff
@@ -39,6 +39,7 @@ function render_block_core_post_template( $attributes, $content, $block ) {
 	if ( ! $query->have_posts() ) {
 		return '';
 	}
+	update_post_thumbnail_cache( $query );
```

Comment:
Since a Query Loop block might not contain the `Post Featured Image` block, wouldn't this result in extra db queries? Can we incorporate this in the Post Featured Image block somehow and have some performance improvements?

Reply:
Fair point. I have moved the logic to the feature image block. 

I tried to get the query as part of context, why I am not sure if this is even possible. My worry is the the feature image is used in a query block or other place, where the query is different than the main query on the page. 

Any ideas @ntsekouras

Reply:
The function `update_post_thumbnail_cache` accepts the query object as a param. When it does, all thumbnail caches are primed in one request. It uses the WP_Query object to get the posts and then loop the thumbnails. 

I think we should pass the query in here, as a feature image block could be used in different query loops.

Reply:
I took a closer look here and I think this could work well in [Post Template](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-template/index.php) block.

```
foreach ( $block->parsed_block['innerBlocks'] as $inner_block ) {
	if ( 'core/post-featured-image' === $inner_block['blockName'] ) {
		update_post_thumbnail_cache( $query );
		break;
	}
}
```

Can you think of any implications with this code @gziolo?

Reply:
@ntsekouras, that would be similar to what @spacedmonkey proposes in https://github.com/WordPress/gutenberg/pull/40752 for the Navigation block.

Anyway, it largely depends on whether the Post Featured Image is the immediate child in the Post Template or it can be included at any level of nesting. I'm not sure whether `$block->parsed_block['innerBlocks']` is a flat list of inner blocks, but it would have to be further investigated if you plan to go this route.

Reply:
>Anyway, it largely depends on whether the Post Featured Image is the immediate child in the Post Template or it can be included at any level of nesting.

Oh.. that's true. It would make sense to have some metrics about this optimization whether it's better to recurse through innerBlocks to find if featured image exists in comparison to the two requests.

Reply:
It would not be 2 requests, if you have 10 feature images on a page, it this could result in 10 times 2 ( one for post one for meta ) for each thumbnail. I have code in [here](https://github.com/WordPress/gutenberg/pull/40752/files#diff-fd9e2a890675c020dd9a7b56054536ce057f6a5dad4de577e1b44625ea617570) to recursively loop through blocks to find nested navigation link blocks. I can confirm, that the inner is, is not a full list. The inner block, have inner blocks. So you have to recursively loop through blocks. It is a pain, but I have hidden the logic, so we could adapt it here. 

The check here would be much simpler, as we just been a boolean of has feature image block. 

 @ntsekouras @gziolo Should I move the logic back to the post template block with a recursively lookup for featured-image blocks?

Reply:
>@ntsekouras @gziolo Should I move the logic back to the post template block with a recursively lookup for featured-image blocks?

Yes, I think it's better to be there and pass the specific query. Also without some metrics I'm not even sure if it worths it to recurse through innerBlocks or just call the function 😄